tests/parsing/FilterHelperTest.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from coalib.parsing.FilterHelper import (
+    apply_filter,
+    apply_filters,
+    is_valid_filter,
+    _filter_section_bears
+)
+from coalib.parsing.filters import available_filters
+from coalib.parsing.InvalidFilterException import InvalidFilterException
+from coalib.settings.ConfigurationGathering import get_all_bears
+
+
+class FilterHelperTest(unittest.TestCase):
+
+    def test_apply_filter_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filter('unknown', ['args'])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_apply_filters_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filters([['language', 'C', 'Python'],
+                           ['unknown', 'args']])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_is_valid_filter(self):
+        wrong_filter_result = is_valid_filter('check')
+        right_filter_result = is_valid_filter('can_detect')
+        self.assertFalse(wrong_filter_result)
+        self.assertTrue(right_filter_result)
+
+    def test_filter_section_bears(self):
+        local_bears, global_bears = get_all_bears()
```

Comment:
For what you are using global `global_bears`? Since I didn't see any use of this variable here, I think you can remove this. :wink:

Reply:
oh. Yeah, I just saw that. If this is the case then it's fine. :+1:

Reply:
You can use `local_bears, _ = get_all_bears()` in this case. But no big deal, this is your choice. 😉